<div style="
    display: grid;
    grid-template-columns: 1fr auto 1fr;
    align-items: center;
    width: 100%;
    margin-top: -10px;">
<!-- Left: logo -->
    <div style="justify-self: start;">
        <img src="2. LOGO HORIZONTAL QUÍMICA.png"
             style="width:330px;">
    </div>

<!-- Center: title -->
<div style="text-align:center;">
        <h1>Fisicoquímica VI : Tarea 1</h1>
        <p style="font-size: 25px;">Diego Méndez</p>
    </div>

<!-- Right empty space -->
<div></div>

</div>


### Tabla de contenidos
* [Tablas y fórmulas](#tablones)
* [Ejercicio 12.3(a)](#ejer1)
* [Ejercicio 12.4(a)](#ejer2)
* [Ejercicio 12.5(a)](#ejer3)
* [Ejercicio 12.6(a)](#ejer4)
* [Ejercicio 12.7(a)](#ejer5)
* [Ejercicio 12.8(a)](#ejer6)
* [Ejercicio 12.9(a)](#ejer7)

<h3 id="tablones">Tablas constantes y fórmulas</h3>

<div style="
    display:flex;
    justify-content:center;
    align-items:flex-start;
    gap:10px;
    flex-wrap:nowrap;
">

  <img src="1.png"
         style="width:60%; height:auto;">

  <img src="2.png"
         style="width:60%; height:auto;">

</div>

<p style="text-align:center;">
Tabla 1. Momentos de inercia. Recuperado de:
Atkins P. & De Paula J.(2010). Physical Chemistry. pp.450
</p>


<div align="center">


 |Constantes| |
|---|---|
|$h$| $6.626 070 15\times 10^{-34}Js$|
|$\hbar$|$1.054 571 817 \times 10^{-34}Js$|
|$N_{A}$|$6.02214076\times 10^{23} \text{mol}^{-1}$|
|$c$|$ 299.792.458 m s^{-1}$|

<p>Tabla 2. Valores de diferentes constantes.</p>

</div>


In [ ]:
import math
import pandas as pd
import numpy as np
from IPython.display import Image
from scipy.constants import Avogadro
from scipy.constants import hbar
from scipy.constants import c

#### **12.3(a)** <a class = "anchor" id="ejer1"></a> 
Calculate the moment of inertia of an $^{31}\text{PH}^{3}$ molecule for rotation
about its threefold axis. By how much does that moment of inertia change
when $^{32}\text{P}$ replaces $^{31}\text{P?}$ (m($^{31}P$) = 30.97mu; Re = 142 pm; HPH angle = 93.6°.)


<div align="center">

<img src="PH3.png"
     alt="Molécula de PH3"
     width="15%">
</div>

\begin{equation}
    I_{\parallel} = 2m_{H} (1-\cos{\theta})R^{2}
    \tag{1}
\end{equation}

In [42]:
uma_h = 1.008
masa_h = uma_h / (1000*Avogadro)
uma_p = 30.97 
masa_p = 30.97/(Avogadro*1000)
angle_rad_12_3 = math.radians(93.6)
r_12_3 = 142*(10**-12)
I_12_3 = 2*masa_h*(1-math.cos(angle_rad_12_3))*r_12_3**2
print(f"I = {I_12_3:.3e} kg m^2")

I = 7.174e-47 kg m^2


Se obtiene $I \approx 7.174\times10^{-47} kg\,m^{2}$
Mientras que si se intercambia $^{31}P$ con $^{32}P$, no habrá cambio con el valor de I, esto es debido a que P no contribuye al momento angular para el eje principal.

#### **12.4(a)** <a class = "anchor" id="ejer2"></a> 

Use the information in Table 12.1 to calculate the moments of inertia and the rotational constants (as frequencies and wavenumbers) of $^{35}Cl^{12}CH_{3}$.

 (m($^{35}Cl$) = 34.9688mu; R(C–H) = 111 pm; R(C–Cl) = 178 pm;
HCH angle = 111°.)

<div align="center">

<img src="CH3Cl.png"
     alt="Molécula de CH3Cl"
     width="15%">
</div>

In [65]:
# Declaración de variables
uma_c = 12.011
uma_cl = 34.9688
r_12_4_1 = 111*(10**-12)
r_12_4_2 = 178*(10**-12)
angle_rad_12_4 = math.radians(111)
#Conversión de unidades
masa_c = uma_c / (1000*Avogadro)
masa_cl = uma_cl / (1000*Avogadro)
masa_total_12_4 = masa_h + masa_c + masa_cl
print(f"Masa total = {masa_total_12_4:.3e} kg")

Masa total = 7.969e-26 kg


#### Expresiones para el momento de inercia:

\begin{equation}
    I_{∥} = 2m_{H} (1-\cos{\theta})R^{2}
    \tag{2}
\end{equation}

\begin{equation}
    I_{\perp} = m_{H}(1-\cos{\theta})R^{2} + \frac{m_{H}}{M}(m_{C}+m_{Cl})(1+2\cos{\theta})R^{2}+ \frac{m_Cl}{M} \{ (3m_{H} + m_{C})R^{'} + 6 m_{H}R [ \frac{1}{3} (1+2 \cos{\theta})]^{1/2} \} R^{'}
    \tag{3}
\end{equation}


In [74]:
#Momento de inercia
I_12_4_1 = 2*(r_12_4_1**2)*masa_h*(1-math.cos(angle_rad_12_4))

a_perp = masa_h*(1-math.cos(angle_rad_12_4))*(r_12_4_1**2)
b_perp = (masa_h/masa_total_12_4)*(masa_c+masa_cl)*(1+2*math.cos(angle_rad_12_4))*(r_12_4_1**2)
c_perp = (masa_cl/masa_total_12_4) * ((
    (3*masa_h + masa_c) * r_12_4_2)+
 6*masa_h*r_12_4_1*((1 + 2*math.cos(angle_rad_12_4))/3)**(0.5))*(r_12_4_2)
I_12_4_2 = a_perp + b_perp + c_perp

In [75]:
print(f"I (Primario) = {I_12_4_1:.3e} kg m^2")
print(f"I (Secundario) = {I_12_4_2:.3e} kg m^2")

I (Primario) = 5.603e-47 kg m^2
I (Secundario) = 6.546e-46 kg m^2


#### Expresiones constantes rotacionales

\begin{equation}
     \widetilde{A}  = \frac{\hbar}{4\pi c I_{∥}}
     \tag{4}
\end{equation}

\begin{equation}
    \widetilde{B} = \frac{\hbar}{4\pi c I_{\perp}}
    \tag{6}
\end{equation}

$\footnotesize{
    \bullet \text{Para obtener A y B (Frecuencias), solamente se retira la constante "\textbf{c}" y se obtiene el resultado en Hz en las constantes rotacionales.}
    }$


In [100]:
#Longitud de onda
cons_a_virgul = hbar/(4*math.pi*c*100*I_12_4_1)
cons_b_virgul = hbar/(4*math.pi*c*100*I_12_4_2)
print(f"A virgul ={cons_a_virgul:.3f} cm^{-1}")
print(f"B virgul ={cons_b_virgul:.3f} cm^{-1}")
# Frecuencias
cons_a_12_4 = cons_a_virgul*c*100
cons_b_12_4 = cons_b_virgul*c*100
print(f"A = {cons_a_12_4:.3e} Hz")
print(f"B = {cons_b_12_4:.3e} Hz")


A virgul =4.996 cm^-1
B virgul =0.428 cm^-1
A = 1.498e+11 Hz
B = 1.282e+10 Hz


Se obtienen:
<center>

$I_{\parallel} = 5.603\times 10^{-47} \, \text{kg m}^{2}$

$I_{\perp} = 6.546\times 10^{-46} \, \text{kg m}^{2}$

$\widetilde{A} = 4.996 cm^{-1}$  \& $A = 1.498 \times 10^{11}$

$\widetilde{B} = 0.428 cm^{-1}$  \& $A = 1.282 \times 10^{10}$
</center>

### **12.5(a)** <a class = "anchor" id="ejer3"></a> 
Calculate the frequency of the J = 4←3 transition in the pure
rotational spectrum of $^{14}\text{N}^{16}\text{O}$. The equilibrium bond length is 115 pm.


Sea $m(^{14}N) = 14.007 \: \text{uma}$ \& $m(^{16}O) = 15.999 \: \text{uma}$

<div align="center">

<img src="NO.png"
     alt="Molécula de NO"
     width="15%">
</div>

$$I = \mu R^2 \; \; \text{con} \; \; \mu = \frac{m_{A}m_{B}}{M}$$
$$B = \frac{\hbar}{4 \pi I}$$
$$ \nu = 2B(J+1)$$



In [48]:
uma_n = 14.007
uma_o = 15.999
masa_n = uma_n / (1000*Avogadro)
masa_o = uma_o / (1000*Avogadro)
masa_total_12_5 = masa_n + masa_o
masa_reducida_12_5 = (masa_n*masa_o) / masa_total_12_5
r_12_5 = 115*(10**-12)

In [49]:
#Momento de inercia
I_12_5 = masa_reducida_12_5*(r_12_5**2)
#Constante rotacional
const_b_12_5 = (hbar)/(4*math.pi* I_12_5)
#Frecuencia
nu_12_5 = 2*const_b_12_5*(3+1)

In [50]:
print(f"B = {const_b_12_5:.3e} Hz")
print(f"v = {nu_12_5:.3e} Hz")

B = 5.117e+10 Hz
v = 4.093e+11 Hz


Sea el valor de la frecuencia tal que:
$$\nu = 4.093\times 10^{11}$$

### **12.6(a)** <a class = "anchor" id="ejer4"></a> 
If the wavenumber of the J = 3←2 rotational transition of $^1\text{H}^{35}\text{Cl}$
considered as a rigid rotator is 63.56 cm−1, what is (a) the moment of inertia
of the molecule, (b) the bond length?

<div align="center">

<img src="HCl.png"
     alt="Molécula de HCl"
     width="15%">
</div>


<div align ="center">
            
$\widetilde{\nu} = 63.56 \; \text{cm}^{-1} $


$63.56 \; \text{cm}^{-1} = 2\widetilde{B}(J+1)$

$\widetilde{B} = \frac{\hbar}{4 \pi I c} \rightarrow I = \frac{\hbar}{ 4 c \pi \widetilde{B}}$

$ I = \mu R^{2} $

</div>

In [51]:
# Momento de inercia
const_b_12_6 = 63.56/6
I_12_6 = (hbar)/(4*math.pi*const_b_12_6*(100)*c)
print(f"I = {I_12_6:.3e} kg m^2")

I = 2.642e-47 kg m^2


In [52]:
# Longitud del enlace
masa_reducida_12_6 = masa_h*masa_cl / (masa_h+masa_cl)
longitud_12_6 = ((I_12_6/masa_reducida_12_6)**(1/2))/1*10**12
print(f"R = {longitud_12_6:.3f} pm")

R = 127.445 pm


Los valores obtenidos son tales que:

$$I=2.642 \times 10^{-47} \text{kg m}^{2}$$

$$R_{\overline{HCl}} = 127.445 \text{ pm}$$


### **12.7(a)** <a class = "anchor" id="ejer5"></a> 
Given that the spacing of lines in the microwave spectrum of $^{27}\text{Al}^1\text{H}$ is
constant at 12.604 $cm^{−1}$, calculate the moment of inertia and bond length of
the molecule.

<div align="center">

<img src="AlH.png"
     alt="Molécula de AlH"
     width="15%">
</div>

m($^{27}Al$) = 26.9815 mu.

$$\Delta \widetilde{\nu} = 12.604 \; \text{cm}^{-1} = 2\widetilde{B}$$

In [53]:
#Momento de inercia
const_b_12_7 = (12.604)/2
I_12_7 = hbar / (4*math.pi*const_b_12_7*(100)*c)
print(f"I = {I_12_7:.3e} kg m^2")
#Longitud del enlace
masa_al = 26.9815 / (1000*Avogadro)
masa_reducida_12_7 = masa_al*masa_h / (masa_al+masa_h)
longitud_12_7 = ((I_12_7/masa_reducida_12_7)**(1/2))/1*10**12
print(f"R = {longitud_12_7:.3f} pm")

I = 4.442e-47 kg m^2
R = 165.918 pm


$$I = 4.442 \times 10^{-47} \text{ kg m}^{2}$$

$$R_{\overline{HAl}} = 165.918 \text{ pm}$$

### **12.8(a)** <a class = "anchor" id="ejer6"></a> 
The rotational constant of $^{127}\text{I}^{35}\text{Cl}$ is $0.1142 \; \text{cm}^{−1}$. 

Calculate the ICl
bond length with m($^{35}\text{Cl}$) = 34.9688mu, m($^{127}\text{I}$) = 126.9045mu.


<div align="center">

<img src="ICl.png"
     alt="Molécula de ICl"
     width="8%">
</div>


$$\widetilde{B} = 0.1142 \; \text{cm}^{-1}$$

Para rotores lineales triatómicos se tiene:

$$I = \mu R ^{2}$$

Donde A, corresponde a los homoátomos que se encuentran en los extremos de cada enlace.


In [54]:
#Momento de inercia
const_b_12_8 = 0.1142
I_12_8 = hbar / (4*math.pi*const_b_12_8*(100)*c)
print(f"I = {I_12_8:.3e} kg m^2")

#Longitud de enlace
masa_I = 126.9045 / (1000*Avogadro)
masa_reducida_12_8 = masa_I*masa_cl / (masa_I+masa_cl)
longitud_12_8 = ((I_12_8/(masa_reducida_12_8))**(1/2))/ 1*10**12
print(f"R = {longitud_12_8:.3f} pm")

I = 2.451e-45 kg m^2
R = 232.046 pm


$$I=2.451 \times 10^{-45} \text{ kg m}^{2}$$

$$R_{\overline{ICl}} = 232.046 \text{ pm}$$

### **12.9(a)** <a class = "anchor" id="ejer7"></a> 
Determine the HC and CN bond lengths in HCN from the rotational
constants B($^{1}\text{H}^{12}\text{C}^{14}\text{N}$) = 44.316 GHz and B($^{2}\text{H}^{12}\text{C}^{14}\text{N}$) = 36.208 GHz.

<div align="center">

<img src="HCN.png"
     alt="Molécula de HCN"
     width="20%">
</div>

In [55]:
const_b_12_9_1 = 44.316 * 10**(9)
const_b_12_9_2 = 36.208 * 10**(9)
#Momentos de inercia
I_12_9_1 = hbar / (4*math.pi*const_b_12_9_1)
I_12_9_2 = hbar / (4*math.pi*const_b_12_9_2)
print(f"I_1 = {I_12_9_1:.3e} kg m^2")
print(f"I_2 = {I_12_9_2:.3e} kg m^2")


I_1 = 1.894e-46 kg m^2
I_2 = 2.318e-46 kg m^2
